### ISSUE TITLE
Uploads stopped here as well since 11/24

### ISSUE THREAD

**Issue by eueyrunbike** [svanhoutte/wyze_garmin_sync#25]
Docker was stuck in a step and i tried stopping and restarting the container to no avail. I tried deleting the container completely and reinstalling it on command prompt. This is what I resulted with. It's stuck at no chksum detected. 

2024-12-06 07:10:10 wyzegarminconnect-1  | [<wyze_sdk.ScaleRecord: {'age': 38, 'bmi': 24.39468, 'bmr': 1630.509, 'body_fat': 17.4, 'body_type': 6, 'body_vfr': 8, 'body_water': 60.4, 'bone_mineral': 3.7, 'device_id': 'JA.SC2.2CAA8E4603EF', 'family_member_id': '5cc3adba9f917c10b86c6662a4ab616a', 'height': 170.0, 'id': '763121048537809419', 'impedance': [592, 486, 455, 426], 'mac': '2CAA8E4603EF', 'measure_ts': 1733426597291, 'measure_type': 1, 'metabolic_age': 35, 'muscle': 54.7, 'occupation': 1, 'protein': 17.0, 'timezone': 'America/Los_Angeles', 'user_id': '5cc3adba9f917c10b86c6662a4ab616a', 'weight': 155.76}>, <wyze_sdk.ScaleRecord: {'age': 38, 'bmi': 24.83665, 'bmr': 1631.5946, 'body_fat': 18.8, 'body_type': 6, 'body_vfr': 8, 'body_water': 59.4, 'bone_mineral': 3.7, 'device_id': 'JA.SC2.2CAA8E4603EF', 'family_member_id': '5cc3adba9f917c10b86c6662a4ab616a', 'height': 170.0, 'id': '5702595969938377898', 'impedance': [608, 501, 465, 433], 'mac': '2CAA8E4603EF', 'measure_ts': 1733325744000, 'metabolic_age': 35, 'muscle': 54.7, 'occupation': 1, 'protein': 16.7, 'timezone': 'America/Los_Angeles', 'user_id': '5cc3adba9f917c10b86c6662a4ab616a', 'weight': 158.58}>]
2024-12-06 07:10:10 wyzegarminconnect-1  | Body Type: 6
2024-12-06 07:10:10 wyzegarminconnect-1  | Generating fit data...
2024-12-06 07:10:10 wyzegarminconnect-1  | Fit data generated...
2024-12-06 07:10:10 wyzegarminconnect-1  | No chksum detected. Uploading fit file and creating chksum...
2024-12-06 07:16:10 wyzegarminconnect-1  | Enter Garmin email address: 2024-12-06T15:16:18.228419441Z mkdir: can't create directory '/etc/cron.d': File exists
2024-12-06 07:16:23 wyzegarminconnect-1  | Scale found with MAC JA.SC2.2CAA8E4603EF. Latest record is:
2024-12-06 07:16:23 wyzegarminconnect-1  | [<wyze_sdk.ScaleRecord: {'age': 38, 'bmi': 24.39468, 'bmr': 1630.509, 'body_fat': 17.4, 'body_type': 6, 'body_vfr': 8, 'body_water': 60.4, 'bone_mineral': 3.7, 'device_id': 'JA.SC2.2CAA8E4603EF', 'family_member_id': '5cc3adba9f917c10b86c6662a4ab616a', 'height': 170.0, 'id': '763121048537809419', 'impedance': [592, 486, 455, 426], 'mac': '2CAA8E4603EF', 'measure_ts': 1733426597291, 'measure_type': 1, 'metabolic_age': 35, 'muscle': 54.7, 'occupation': 1, 'protein': 17.0, 'timezone': 'America/Los_Angeles', 'user_id': '5cc3adba9f917c10b86c6662a4ab616a', 'weight': 155.76}>, <wyze_sdk.ScaleRecord: {'age': 38, 'bmi': 24.83665, 'bmr': 1631.5946, 'body_fat': 18.8, 'body_type': 6, 'body_vfr': 8, 'body_water': 59.4, 'bone_mineral': 3.7, 'device_id': 'JA.SC2.2CAA8E4603EF', 'family_member_id': '5cc3adba9f917c10b86c6662a4ab616a', 'height': 170.0, 'id': '5702595969938377898', 'impedance': [608, 501, 465, 433], 'mac': '2CAA8E4603EF', 'measure_ts': 1733325744000, 'metabolic_age': 35, 'muscle': 54.7, 'occupation': 1, 'protein': 16.7, 'timezone': 'America/Los_Angeles', 'user_id': '5cc3adba9f917c10b86c6662a4ab616a', 'weight': 158.58}>]
2024-12-06 07:16:23 wyzegarminconnect-1  | Body Type: 6
2024-12-06 07:16:23 wyzegarminconnect-1  | Generating fit data...
2024-12-06 07:16:23 wyzegarminconnect-1  | Fit data generated...
2024-12-06 07:16:23 wyzegarminconnect-1  | No chksum detected. Uploading fit file and creating chksum...

**Comment by svanhoutte** [comment]
Did you upgrade to the last version of the container ? ensure that the docker-compose file has image: svanhoutte/wyzegarminconnect:latest, then docker compose pull.
Restart the docker and let me know.

**Comment by eueyrunbike** [comment]
Hi, I checked the version in the compose file and it is version 3.9 and says latest. Upgraded to the latest version of docker and deleted the container again and "compose up -d" and "compose log -f " in command prompt. it gets stuck after no chksum detected. It's strange. I don't touch anything after I get it up and running. The computer goes to sleep sometimes and there are times I'm not at it for a couple days. I don't change anything with networking or really anything. I keep running into little issues where it stops uploading like this issue. It's all good when it works. I just get confused on when it stops working at times. Here is what I get at the command prompt:
C:\Users\Monica\.docker>docker compose up -d
time="2024-12-06T08:26:30-08:00" level=warning msg="C:\\Users\\Monica\\.docker\\docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion"
[+] Running 1/1
 ✔ Container docker-wyzegarminconnect-1  Started                                                                   0.7s

C:\Users\Monica\.docker>docker compose logs -f
time="2024-12-06T08:26:42-08:00" level=warning msg="C:\\Users\\Monica\\.docker\\docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion"
wyzegarminconnect-1  | Scale found with MAC JA.SC2.2CAA8E4603EF. Latest record is:
wyzegarminconnect-1  | [<wyze_sdk.ScaleRecord: {'age': 38, 'bmi': 24.39468, 'bmr': 1630.509, 'body_fat': 17.4, 'body_type': 6, 'body_vfr': 8, 'body_water': 60.4, 'bone_mineral': 3.7, 'device_id': 'JA.SC2.2CAA8E4603EF', 'family_member_id': '5cc3adba9f917c10b86c6662a4ab616a', 'height': 170.0, 'id': '763121048537809419', 'impedance': [592, 486, 455, 426], 'mac': '2CAA8E4603EF', 'measure_ts': 1733426597291, 'measure_type': 1, 'metabolic_age': 35, 'muscle': 54.7, 'occupation': 1, 'protein': 17.0, 'timezone': 'America/Los_Angeles', 'user_id': '5cc3adba9f917c10b86c6662a4ab616a', 'weight': 155.76}>, <wyze_sdk.ScaleRecord: {'age': 38, 'bmi': 24.83665, 'bmr': 1631.5946, 'body_fat': 18.8, 'body_type': 6, 'body_vfr': 8, 'body_water': 59.4, 'bone_mineral': 3.7, 'device_id': 'JA.SC2.2CAA8E4603EF', 'family_member_id': '5cc3adba9f917c10b86c6662a4ab616a', 'height': 170.0, 'id': '5702595969938377898', 'impedance': [608, 501, 465, 433], 'mac': '2CAA8E4603EF', 'measure_ts': 1733325744000, 'metabolic_age': 35, 'muscle': 54.7, 'occupation': 1, 'protein': 16.7, 'timezone': 'America/Los_Angeles', 'user_id': '5cc3adba9f917c10b86c6662a4ab616a', 'weight': 158.58}>]
wyzegarminconnect-1  | Body Type: 6
wyzegarminconnect-1  | Generating fit data...
wyzegarminconnect-1  | Fit data generated...
wyzegarminconnect-1  | No chksum detected. Uploading fit file and creating chksum...

Thanks!

**Comment by svanhoutte** [comment]
lets try that, please confirm you are not running any other container with docker: 
docker compose down 
docker system prune 
docker compose pull 
docker compose run --rm wyzegarminconnect 
(exit the container CTRL+C)
docker compose up -d

**Comment by eueyrunbike** [comment]
Ok, I tried to follow along as best I know how. I was on my phone reading the email so saw the "wyzegarminconnect" part on the next line thus my error in entering it in command prompt... but after all said and done, I'm left at the same "No chksum detected. Uploading fit file and creating chksum..."

C:\Users\Monica>cd .docker

C:\Users\Monica\.docker>docker compose down
time="2024-12-06T09:02:51-08:00" level=warning msg="C:\\Users\\Monica\\.docker\\docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion"
[+] Running 1/1
 ✔ Container docker-wyzegarminconnect-1  Removed                                                                  10.4s

C:\Users\Monica\.docker>docker system prune
WARNING! This will remove:
  - all stopped containers
  - all networks not used by at least one container
  - all dangling images
  - unused build cache

Are you sure you want to continue? [y/N] y
Total reclaimed space: 0B

C:\Users\Monica\.docker>docker compose pull
time="2024-12-06T09:03:42-08:00" level=warning msg="C:\\Users\\Monica\\.docker\\docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion"
[+] Pulling 1/1
 ✔ wyzegarminconnect Pulled                                                                                        1.9s

C:\Users\Monica\.docker>docker compose run --rm
requires at least 1 arg(s), only received 0

C:\Users\Monica\.docker>docker compose run--rm

Usage:  docker compose [OPTIONS] COMMAND

Define and run multi-container applications with Docker

Options:
      --all-resources              Include all resources, even those not
                                   used by services
      --ansi string                Control when to print ANSI control
                                   characters ("never"|"always"|"auto")
                                   (default "auto")
      --compatibility              Run compose in backward compatibility mode
      --dry-run                    Execute command in dry run mode
      --env-file stringArray       Specify an alternate environment file
  -f, --file stringArray           Compose configuration files
      --parallel int               Control max parallelism, -1 for
                                   unlimited (default -1)
      --profile stringArray        Specify a profile to enable
      --progress string            Set type of progress output (auto,
                                   tty, plain, json, quiet) (default "auto")
      --project-directory string   Specify an alternate working directory
                                   (default: the path of the, first
                                   specified, Compose file)
  -p, --project-name string        Project name

Commands:
  attach      Attach local standard input, output, and error streams to a service's running container
  build       Build or rebuild services
  config      Parse, resolve and render compose file in canonical format
  cp          Copy files/folders between a service container and the local filesystem
  create      Creates containers for a service
  down        Stop and remove containers, networks
  events      Receive real time events from containers
  exec        Execute a command in a running container
  export      Export a service container's filesystem as a tar archive
  images      List images used by the created containers
  kill        Force stop service containers
  logs        View output from containers
  ls          List running compose projects
  pause       Pause services
  port        Print the public port for a port binding
  ps          List containers
  pull        Pull service images
  push        Push service images
  restart     Restart service containers
  rm          Removes stopped service containers
  run         Run a one-off command on a service
  scale       Scale services
  start       Start services
  stats       Display a live stream of container(s) resource usage statistics
  stop        Stop services
  top         Display the running processes
  unpause     Unpause services
  up          Create and start containers
  version     Show the Docker Compose version information
  wait        Block until containers of all (or specified) services stop.
  watch       Watch build context for service and rebuild/refresh containers when files are updated

Run 'docker compose COMMAND --help' for more information on a command.
unknown docker command: "compose run--rm"

C:\Users\Monica\.docker>

C:\Users\Monica\.docker>wyzegarminconnect
'wyzegarminconnect' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\Monica\.docker>docker compose run --rm wyzegarminconnect
time="2024-12-06T09:06:02-08:00" level=warning msg="C:\\Users\\Monica\\.docker\\docker-compose.yml: the attribute `version` is obsolete, it will be ignored, please remove it to avoid potential confusion"
Scale found with MAC JA.SC2.2CAA8E4603EF. Latest record is:
[<wyze_sdk.ScaleRecord: {'age': 38, 'bmi': 24.39468, 'bmr': 1630.509, 'body_fat': 17.4, 'body_type': 6, 'body_vfr': 8, 'body_water': 60.4, 'bone_mineral': 3.7, 'device_id': 'JA.SC2.2CAA8E4603EF', 'family_member_id': '5cc3adba9f917c10b86c6662a4ab616a', 'height': 170.0, 'id': '763121048537809419', 'impedance': [592, 486, 455, 426], 'mac': '2CAA8E4603EF', 'measure_ts': 1733426597291, 'measure_type': 1, 'metabolic_age': 35, 'muscle': 54.7, 'occupation': 1, 'protein': 17.0, 'timezone': 'America/Los_Angeles', 'user_id': '5cc3adba9f917c10b86c6662a4ab616a', 'weight': 155.76}>, <wyze_sdk.ScaleRecord: {'age': 38, 'bmi': 24.83665, 'bmr': 1631.5946, 'body_fat': 18.8, 'body_type': 6, 'body_vfr': 8, 'body_water': 59.4, 'bone_mineral': 3.7, 'device_id': 'JA.SC2.2CAA8E4603EF', 'family_member_id': '5cc3adba9f917c10b86c6662a4ab616a', 'height': 170.0, 'id': '5702595969938377898', 'impedance': [608, 501, 465, 433], 'mac': '2CAA8E4603EF', 'measure_ts': 1733325744000, 'metabolic_age': 35, 'muscle': 54.7, 'occupation': 1, 'protein': 16.7, 'timezone': 'America/Los_Angeles', 'user_id': '5cc3adba9f917c10b86c6662a4ab616a', 'weight': 158.58}>]
Body Type: 6
Generating fit data...
Fit data generated...
No chksum detected. Uploading fit file and creating chksum...
Enter Garmin email address: 
Enter Garmin password:
GarthHTTPError(msg='Error in request', error=HTTPError('401 Client Error: Unauthorized for url: https://sso.garmin.com/sso/signin?id=gauth-widget&embedWidget=true&gauthHost=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&service=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&source=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&redirectAfterAccountLoginUrl=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed&redirectAfterAccountCreationUrl=https%3A%2F%2Fsso.garmin.com%2Fsso%2Fembed'))
crond[14]: crond (busybox 1.36.1) started, log level 8
docker compose run --rm wyzegarminconnect
^C
C:\Users\Monica\.docker>

C:\Users\Monica\.docker>docker compose up-d

Usage:  docker compose [OPTIONS] COMMAND

Define and run multi-container applications with Docker

Options:
      --all-resources              Include all resources, even those not
                                   used by services
      --ansi string                Control when to print ANSI control
                                   characters ("never"|"always"|"auto")
                                   (default "auto")
      --compatibility              Run compose in backward compatibility mode
      --dry-run                    Execute command in dry run mode
      --env-file stringArray       Specify an alternate environment file
  -f, --file stringArray           Compose configuration files
      --parallel int               Control max parallelism, -1 for
                                   unlimited (default -1)
      --profile stringArray        Specify a profile to enable
      --progress string            Set type of progress output (auto,
                                   tty, plain, json, quiet) (default "auto")
      --project-directory string   Specify an alternate working directory
                                   (default: the path of the, first
                                   specified, Compose file)
  -p, --project-name string        Project name

Commands:
  attach      Attach local standard input, output, and error streams to a service's running container
  build       Build or rebuild services
  config      Parse, resolve and render compose file in canonical format
  cp          Copy files/folders between a service container and the local filesystem
  create      Creates containers for a service
  down        Stop and remove containers, networks
  events      Receive real time events from containers
  exec        Execute a command in a running container
  export      Export a service container's filesystem as a tar archive
  images      List images used by the created containers
  kill        Force stop service containers
  logs        View output from containers
  ls          List running compose projects
  pause       Pause services
  port        Print the public port for a port binding
  ps          List containers
  pull        Pull service images
  push        Push service images
  restart     Restart service containers
  rm          Removes stopped service containers
  run         Run a one-off command on a service
  scale       Scale services
  start       Start services
  stats       Display a live stream of container(s) resource usage statistics
  stop        Stop services
  top         Display the running processes
  unpause     Unpause services
  up          Create and start containers
  version     Show the Docker Compose version information
  wait        Block until containers of all (or specified) services stop.
  watch       Watch build context for service and rebuild/refresh containers when files are updated

Run 'docker compose COMMAND --help' for more information on a command.
unknown docker command: "compose up-d"

[+] Running 1/1\.docker>docker compose up -d
 ✔ Container docker-wyzegarminconnect-1  Started                                                                   0.7s
on` is obsolete, it will be ignored, please remove it to avoid potential confusion"
C:\Users\Monica\.docker>

**Comment by svanhoutte** [comment]
it seems you are having an authentication issue with Garmin. 
shutdown the container, delete what is in the tokens directory then run "docker compose run --rm wyzegarminconnect"
it should ask for the MFA that you will receive by email, enter it and you should be good after that. 
Exit the container (CTRL+C) then docker compose up -d

(I edited your previous comment to remove your email address.)

**Comment by eueyrunbike** [comment]
Hey! I just got time to follow up on this and in the process with the docker-compose read me file open, my garmin password caught my eye and I thought I could've changed that recently. I quickly try signing on to Garmin connect and that password was wrong as I indeed have changed it. User error dammit! But I finished up with your instructions since I had already started the process and deleted what was in the tokens directory. Thanks man. It's back up and running!
